main.tf

```diff
@@ -19,3 +19,15 @@ resource "aws_sqs_queue" "this" {
 
   tags = var.tags
 }
+
+data "aws_arn" "this" {
+  count = var.create ? 1 : 0
+
+  arn = element(
```

Comment:
This can be updated to - `arn = aws_sqs_queue.this[0].arn`

There is no need in `element(concat(...))` when you use `count`.

Reply:
Ok. What is the point to use this scheme with `concat()` in outputs.tf?

Reply:
There is no `count` inside of `output`, so we need to find the first non-empty value there and can't reference by index `[0]` as we can do in data-sources or resources with `count`.

Reply:
Got it! Thanks!